src/argilla/client/sdk/datasets/v1/api.py

```diff
@@ -0,0 +1,152 @@
+def get_dataset(
+    client: AuthenticatedClient,
+    id: str,
+) -> Response[Union[FeedbackDataset, ErrorMessage, HTTPValidationError]]:
+    url = "{}/api/v1/datasets/{id}".format(client.base_url, id=id)
+
+    response = httpx.get(
+        url=url,
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+
+    if response.status_code == 200:
+        parsed_response = FeedbackDataset(**response.json())
+        return Response(
+            status_code=response.status_code,
+            content=response.content,
+            headers=response.headers,
+            parsed=parsed_response,
+        )
+    return handle_response_error(response)
+
+
+def publish_dataset(
+    client: AuthenticatedClient,
+    id: str,
+) -> Response[Union[FeedbackDataset, ErrorMessage, HTTPValidationError]]:
+    url = "{}/api/v1/datasets/{id}/publish".format(client.base_url, id=id)
+
+    response = httpx.put(
+        url=url,
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+
+    if response.status_code == 200:
+        parsed_response = FeedbackDataset(**response.json())
+        return Response(
+            status_code=response.status_code,
+            content=response.content,
+            headers=response.headers,
+            parsed=parsed_response,
+        )
+    return handle_response_error(response)
+
+
+def list_datasets(
+    client: AuthenticatedClient,
+) -> Response[Union[List[FeedbackDataset], ErrorMessage, HTTPValidationError]]:
+    url = "{}/api/v1/datasets".format(client.base_url)
+
+    response = httpx.get(
+        url=url,
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+
+    if response.status_code == 200:
+        parsed_response = [FeedbackDataset(**dataset) for dataset in response.json()["items"]]
+        return Response(
+            status_code=response.status_code,
+            content=response.content,
+            headers=response.headers,
+            parsed=parsed_response,
+        )
+    return handle_response_error(response)
+
+
+def get_records(
+    client: AuthenticatedClient,
+    id: str,
+    offset: int = 0,
+    limit: int = 50,
+) -> Response[Union[Dict[str, Any], ErrorMessage, HTTPValidationError]]:
+    url = "{}/api/v1/datasets/{id}/records".format(client.base_url, id=id)
+
+    response = httpx.get(
+        url=url,
+        params={"offset": offset, "limit": limit},
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+
+    if response.status_code == 200:
+        return Response(
+            status_code=response.status_code,
+            content=response.content,
+            headers=response.headers,
+            parsed=response.json(),
+        )
+    return handle_response_error(response)
+
+
+def add_record(
```

Comment:
Also it seems that we're no longer able to add records until the dataset is published, is this correct? @jfcalvo

Reply:
Exactly. Right now the flow should be the following:
1. Create the Dataset.
2. Add at least one Field to the Dataset.
3. Add at least one Question to the Dataset.
4. Publish the Dataset.
5. Add records to the dataset that matches the defined fields and questions for the dataset.

Before publishing the dataset is not possible to add records.

Reply:
Cool, I'll push both the `fields` and `questions` methods today to cover all the `/api/v1` endpoints 😄